packages/common/src/utils/mapObject.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Map each key of a source object via a given valueMap function
+ */
+export function mapObject<
+  Source extends Record<string | number | symbol, unknown>,
+  Target extends { [key in keyof Source]: unknown }
+>(source: Source, valueMap: (key: keyof Source, value: Source[typeof key]) => Target[typeof key]): Target {
```

Comment:
slight preference for reversing the key/value order here since you most often want to mutate the value and can ignore the second key param, but not sure if that makes it harder to type `value` because it depends on `key`

Reply:
updated!